qutip/qobj.py

```diff
@@ -2018,6 +2020,28 @@ def isunitary(self):
     def isunitary(self, isunitary):
         self._isunitary = isunitary
 
+    def check_isdensity(self):
+        """
+        Checks whether qobj is a valid density matrix
+        """
+        isdensity = True
+
+        if (self.type != "oper" or self.isherm is False or
+                np.allclose(self.tr(), 1, atol=1e-09) is False):
+            isdensity = False
+
+        else:
+            for eig in self.eigenenergies():
```

Comment:
We can make this a lot faster in some cases by only requesting the lowest eigenvalue.